functions/index.js

```diff
@@ -12,6 +13,32 @@ admin.initializeApp(functions.config().firebase);
 exports.createNewUser = functions.auth.user().onCreate(event => {
   const uid = event.data.uid;
   const email = event.data.email;
-  
+
   return admin.database().ref(`/users/${uid}`).set({ email });
 });
+
+exports.joinTheClub = functions.database.ref('/queues/joinTheClub/tasks/{taskId}').onWrite(event => {
+  if (event.data._newData) {
+    const { inviteCode, userId } = event.data._newData;
+
+    return admin.database().ref('bookClub').orderByChild('inviteCode').equalTo(inviteCode).once('value').then((response) => {
+      if (response.exists() && response.val()) {
+        const val = response.val();
+        const bookClubId = Object.keys(val)[0];
+        const bookClubName = Object.keys(val)[0].name;
+        const randomId = Math.random().toString(36).slice(-10);
+
+        const updates = {
+          [`bookClub/${bookClubId}/users/${userId}`]: true,
+          [`users/${userId}/bookClubs/${bookClubId}`]: true,
+          [`queues/joinTheClub/tasks/${event.params.taskId}`]: null,
+          [`users/${userId}/notifications/${randomId}`]: { message: `You were successfully added to ${bookClubName}` }
+        };
+        return admin.database().ref().update(updates);
+      } else {
+        admin.database().ref('queues/joinTheClub/tasks').child(event.params.taskId).remove();
+        return admin.database().ref(`users/${userId}/notifications`).push({ message: 'The invite code you submitted didn\'t work' });
```

Comment:
I'm not sure exactly how it works, but it does need an id in order to go into firebase database. In the `if` part of the statement, I had to generate the notificationId myself since I couldn't do a `push()` while also doing a giant `update()`. But in this `else` statement, all I'm changing in the database is the warning that it failed (it didn't find a book club with the inviteCode you provided), so all I need is the `push()`